### PR TITLE
Increase the supported versions for MyBatis-Plus. 3.0~3.1

### PR DESCRIPTION
**Pull request by homejim**
Increase the supported versions for MyBatis-Plus. After the increase, the supported versions range from 3.0.0 to 3.5.8

### CODE REVIEW

**Comment by homejim** [comment]
@skybber cr

File: plugin/hotswap-agent-mybatis-plus-plugin/src/main/java/org/hotswap/agent/plugin/mybatisplus/transformers/MyBatisPlusTransformers.java [hunk 1]
```diff
@@ -116,53 +133,94 @@ public static void patchPlusSqlSessionFactoryBuilder(CtClass ctClass, ClassPool
...
+            String wrapperCode = "{" +
+                    "if (this." + FACTORYBEAN_FIELD + " != null) {" +
+                    "$1 = " + PlusSqlSessionFactoryBeanCaller.class.getName() + ".proxyPlusConfiguration(this." + FACTORYBEAN_FIELD + ", $1);" +
+                    "}" +
+                    "}";
+            AtomicBoolean existCast = new AtomicBoolean(false);
+            buildMethod.instrument(new ExprEditor() {
+                @Override
+                public void edit(Cast c) throws CannotCompileException {
+                    // 判断是否是 MybatisConfiguration 的强制转换
```
Comment (skybber) [inline plugin/hotswap-agent-mybatis-plus-plugin/src/main/java/org/hotswap/agent/plugin/mybatisplus/transformers/MyBatisPlusTransformers.java:171]: please english comments

**Comment by homejim** [comment]
@skybber The useless comment was removed

**Comment by homejim** [comment]
@skybber Additionally, this is unrelated to the current commit. Whether it's previously released code or new code, running `mvn test` always fails due to the `hotswap-agent-proxy-plugin plugin`, both on Windows and Linux. The logs are available here: https://github.com/homejim/HotswapAgent/actions/runs/11136702281/job/30948925103.

Currently, when I run unit tests, I need to comment out  `hotswap-agent-proxy-plugin plugin` plugin test first.

Is there a parameter that can be added to make it run correctly?

**Comment by skybber** [comment]
> @skybber Additionally, this is unrelated to the current commit. Whether it's previously released code or new code, running `mvn test` always fails due to the `hotswap-agent-proxy-plugin plugin`, both on Windows and Linux. The logs are available here: https://github.com/homejim/HotswapAgent/actions/runs/11136702281/job/30948925103.
> 
> Currently, when I run unit tests, I need to comment out `hotswap-agent-proxy-plugin plugin` plugin test first.
> 
> Is there a parameter that can be added to make it run correctly?

Tests fails on jbr17/21 since `java.lang.Proxy` was modified between java11->java17. Would be nice to have fixed it in jbr17/21 as well. I'm compiling HA with Trava11.

**Comment by skybber** [comment]
Thanks for your great work!

**Pull request merged**